component/otelcol/receiver/jaeger/jaeger_test.go

```diff
@@ -0,0 +1,120 @@
+package jaeger_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component/otelcol/receiver/jaeger"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/grafana/agent/pkg/util"
+	"github.com/phayes/freeport"
+	"github.com/stretchr/testify/require"
+)
+
+// Test ensures that otelcol.receiver.jaeger can start successfully.
+func Test(t *testing.T) {
+	httpAddr := getFreeAddr(t)
+
+	ctx := componenttest.TestContext(t)
+	l := util.TestLogger(t)
+
+	ctrl, err := componenttest.NewControllerFromID(l, "otelcol.receiver.jaeger")
+	require.NoError(t, err)
+
+	cfg := fmt.Sprintf(`
+		protocols {
+			grpc {
+				endpoint = "%s"
+			}
+		}
+
+		output { /* no-op */ }
+	`, httpAddr)
+	var args jaeger.Arguments
+	require.NoError(t, river.Unmarshal([]byte(cfg), &args))
+
+	go func() {
+		err := ctrl.Run(ctx, args)
+		require.NoError(t, err)
+	}()
+
+	require.NoError(t, ctrl.WaitRunning(time.Second))
+
+	// TODO(rfratto): is it worthwhile trying to make sure we can send data over
+	// the client or can we trust that getting the component to run successfully
+	// is enough?
+	time.Sleep(100 * time.Millisecond)
```

Comment:
WDYT @mattdurham @tpaschalis?

Reply:
I think in general we should be testing the component wrapping changes and not the underlying object, and it would be nice to test the underlying object but not necessary. So I think the above looks reasonable.